sdk/python/tests/test_read.py

```diff
@@ -1,37 +1,148 @@
 """ Test read functions """
 
-from pvsite_datamodel.read.site import get_site, get_all_site, get_site_from_uuid
-from pvsite_datamodel.read.generation import get_pv_generation
+import uuid
+
+from sqlalchemy.orm import Query
+from typing import List
+
+from pvsite_datamodel.sqlmodels import SiteSQL, StatusSQL
+from pvsite_datamodel.read.site import get_all_sites
+from pvsite_datamodel.read.site import get_site_by_uuid
+from pvsite_datamodel.read.site import get_site_by_client_site_id
+from pvsite_datamodel.read.generation import get_pv_generation_by_sites
+from pvsite_datamodel.read.status import get_latest_status
+from pvsite_datamodel.read.generation import get_pv_generation_by_client
+from pvsite_datamodel.read.latest_forecast_values import get_latest_forecast_values_by_site
+
 import pytest
 
 
-def test_get_site(sites, db_session):
+class TestGetAllSites:
+    """Tests for the get_all_sites function"""
+
+    def test_returns_all_sites(self, sites, db_session):
+        out = get_all_sites(db_session)
+
+        assert len(out) == len(sites)
+        assert out[0] == sites[0]
+
+
+class TestGetSiteByUUID:
```

Comment:
nice, i like this grouping

Reply:
Bit easier to see in terms of coverage what has and hasn't been done this way, as well as enabling idiomatic test naming so it's clear what the tests themselves are doing